src/zope/file/upload.py

```diff
@@ -160,6 +162,10 @@ def updateBlob(ob, input):
     else:
         ob.mimeType = mimeType
         ob.parameters = {}
+
     w = ob.open("w")
-    w.write(data)
-    w.close()
+    try:
+        # if we're security proxied, may not be able to use a `with`
```

Comment:
Oh my.  That's very unfortunate!  And something to be fixed (separately from this PR).